cmd/ssh/ssh.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/oklog/run"
+	"github.com/sourcepods/sourcepods/cmd"
+	"github.com/sourcepods/sourcepods/pkg/gitssh"
+	"github.com/sourcepods/sourcepods/pkg/storage"
+	jaeger "github.com/uber/jaeger-client-go"
+	"github.com/uber/jaeger-client-go/config"
+	"github.com/urfave/cli"
+)
+
+type sshConf struct {
+	SSHPort        int
+	HostKeyPath    string
+	StorageGRPCURL string
+	HTTPAddr       string
+	LogJSON        bool
+	LogLevel       string
+	TracingURL     string
+}
+
+var (
+	sshConfig = sshConf{}
+	sshFlags  = []cli.Flag{
+		cli.IntFlag{
+			Name:        cmd.FlagSSHPort,
+			EnvVar:      cmd.EnvSSHPort,
+			Value:       22,
+			Usage:       "The SSH port to listen on",
+			Destination: &sshConfig.SSHPort,
+		},
+		cli.StringFlag{
+			Name:        cmd.FlagSSHHostKeyPath,
+			EnvVar:      cmd.EnvSSHHostKeyPath,
+			Value:       "/etc/ssh/",
+			Usage:       "The path to looks for ssh host-keys in",
+			Destination: &sshConfig.HostKeyPath,
+		},
+		cli.StringFlag{
+			Name:        cmd.FlagStorageGRPCURL,
+			EnvVar:      cmd.EnvStorageGRPCURL,
+			Usage:       "The storage's grpc url to connect with",
+			Destination: &sshConfig.StorageGRPCURL,
+		},
+		cli.BoolFlag{
+			Name:        cmd.FlagLogJSON,
+			EnvVar:      cmd.EnvLogJSON,
+			Usage:       "The logger will log json lines",
+			Destination: &sshConfig.LogJSON,
+		},
+		cli.StringFlag{
+			Name:        cmd.FlagLogLevel,
+			EnvVar:      cmd.EnvLogLevel,
+			Usage:       "The log level to filter logs with before printing",
+			Value:       "info",
+			Destination: &sshConfig.LogLevel,
+		},
+		cli.StringFlag{
```

Comment:
This is currently unused, but to do `repo-name => repo-id`-mapping we would need to "access the DB somehow" :unamused: